### ISSUE TITLE
Using SASS or LESS instead of normal CSS for javafx styling 

### ISSUE THREAD

**Issue by madoar** [PlayOnLinux/POL-POM-5#672]
I've thought a bit about how to set the urls directly inside the css-stylesheets in a nice and pretty way.
My result is, that it's quite hard. One of the reasons is, that it's not possible to define custom css function for css, which, as an example, would be nice in a case like: `-fx-background-image: url(@theme-root, "icons/mainwindow/settings/userInterface.png")`, where `@theme-root` is a variable, containing a path.
One way to solve this, and possible make the stylesheets better readable, would be to move from "normal" CSS to SASS or LESS, which can be transpiled/preprocessed to CSS.
As an example, with less it's possible to define both variables and functions/mixins, which my example above would need.
The only bad part of LESS or SASS is, that it needs to be transpiled to CSS, before javafx can use it. To do this I see two possible methods:

- The first way would be to transpile them inside the application during runtime, whenever we're changing the stylesheet ([see](https://tomsondev.bestsolution.at/2013/08/07/using-less-in-javafx/) for an example)
- The other way would be to write/use a maven plugin, that transpiles the LESS/SASS file for us during the compilation phase

What do you think about this?

**Comment by plata** [comment]
This is only useful if we support themes which are not in the classpath, right?

**Comment by madoar** [comment]
In my mind this would make it much easier to move the image setting parts from the source code to the css files. 
Because this would enable us to directly copy the file paths from the source code to the css files.
Another benefit is, that this approach is not as error prone as using normal css, because we can set real variables for a lot of things, not only for color values.
What I would like a lot too is the possible to wrap elements inside each other, which enables us to make the flat css files more readable.

**Comment by plata** [comment]
Currently, it would be e.g.:
```CSS
-fx-background-image: url('/org/phoenicis/javafx/themes/default/icons/mainwindow/apps/custom.png');
```
I admit that replacing `/org/phoenicis/javafx/themes/default` by a variable is nicer.

Could you maybe prepare a PR showing the different advantages of SASS/LESS compared to just CSS and the impact this would have on the build etc.?

**Comment by madoar** [comment]
The change to less/sass would go hand in hand with the move of the image definitions from the source code to the css files.
Because this is a lot of work, I will start by making a PR, where I simply move one component from the source code to the css file (maybe the SearchBox). In this PR I'll use Less, which will display how we can use less with POL 5.
The question I need to ask before I start is, what phase would you prefer to preprocess the less files to css. At runtime or at compile time?

**Comment by plata** [comment]
Sure. Don't do the whole thing before we agreed/disagreed based on a small example. I would definitely do it at compile time.

**Comment by madoar** [comment]
Ok, I've created a Pull Request, illustrating what I want to do at a really small part of our gui, the search box component, see #677 for more details.

**Comment by plata** [comment]
Fixed by #788.
